lib/services/subscription/service.js

```diff
@@ -334,3 +388,288 @@ function securityDirectoryChanged(whatHappnd, _changedData, effectedSessions) {
+      sub.data.action,
+      '*' // sub.data.searchPath'*'
+    );
+  });
+}
+
+function __removeExplicitlyRevokedSubscriptions(effectedSession, allowedSubs, prohibitedPaths) {
+  let revoked = allowedSubs.filter(
+    sub =>
+      prohibitedPaths.explicit.includes(sub.data.searchPath) ||
+      prohibitedPaths.wild.some(path => this.__startsWithMask(sub.data.searchPath, path))
+  );
+  revoked.forEach(sub => {
+    this.removeListenerExtended(
+      {
+        data: {
+          ref: sub.data.ref,
+          options: { prohibited: { $ne: true } },
+          searchPath: sub.data.searchPath
+        }
+      },
+      effectedSession.id,
+      sub.data.action,
+      '*' // sub.data.searchPath
+    );
+  });
+}
+
+function __addNewProhibitions(effectedSession, prohibitedSubs, prohibitedPaths, wildSubPaths) {
+  let wildProhibitedSubsPaths = prohibitedSubs
+    .filter(sub => sub.data.searchPath.endsWith('*'))
+    .map(sub => sub.data.searchPath.replace(/\*+$/, ''));
+  let newProhibitions = prohibitedPaths.explicit.filter(
+    prohibitPath =>
+      wildSubPaths.some(subPath => this.__startsWithMask(prohibitPath, subPath)) &&
+      !prohibitedSubs.some(sub => sub.data.searchPath === prohibitPath) &&
+      !wildProhibitedSubsPaths.some(path => this.__startsWithMask(prohibitPath, path))
+  );
+  //Need to add a reference for these prohibitions (?)
+  newProhibitions.forEach(path => {
+    this.addListener('ALL', path, effectedSession.id, {
+      options: { prohibited: true },
+      session: effectedSession
+    });
+  });
+}
+
+function __addNewSubscriptions(
+  effectedSession,
+  allowedSubs,
+  wildSubs,
+  allowedPaths,
+  prohibitedPaths,
+  wildSubPaths
+) {
+  //Add subscriptions on paths where we have a wild subscription further up the tree, and now have a permission
+  let allowedSubPaths = allowedSubs.map(sub => sub.data.searchPath);
+  let newSubPaths = allowedPaths.explicit.filter(
+    allowedPath =>
+      !allowedPath.endsWith('*') &&
+      allowedSubPaths.indexOf(allowedPath) < 0 &&
+      wildSubPaths.some(subPath => this.__startsWithMask(allowedPath, subPath)) &&
+      prohibitedPaths.explicit.indexOf(allowedPath) < 0 &&
+      !prohibitedPaths.wild.some(prohibitedPath =>
+        this.__startsWithMask(allowedPath, prohibitedPath)
+      )
+  );
+  newSubPaths.forEach(newPath => {
+    wildSubs
+      .filter(wildSub => this.__startsWithMask(newPath, wildSub.data.path.replace(/\/\*+$/, '/')))
+      .forEach(sub => {
+        let action = sub.data.action;
+        this.addListener(
+          action,
+          newPath,
+          effectedSession.id,
+          {
+            ...sub.data,
+            options: {
+              ...sub.data.options,
+              wild: false
+            },
+            searchPath: newPath
+          },
+          sub.data.path
+        );
+      });
+  });
+}
+
+function __processUnnestedSubscriptions(effectedSession, sessionCB) {
+  try {
+    //TODO: filter out salient paths here
```

Comment:
Is this something that still needs to happen?

Reply:
Yes. If nested subscriptions are switched off, we rather do this because it's a lot less moving parts.